smdebug/tensorflow/keras.py

```diff
@@ -479,7 +496,9 @@ def _on_any_batch_begin(self, batch, mode, logs=None):
             self._prepare_collections()
 
         if self._prepared_tensors[mode] is False:
-            if self._validate_exec_function(self._get_exec_function(mode)):
+            if (is_tf_version_2x() and tf.executing_eagerly()) or self._validate_exec_function(
+                self._get_exec_function(mode)
```

Comment:
this can be None as you commented above?

Reply:
get exec function can be None.

Reply:
so this will result to self._validate_exec_function(None), is this valid? 
In line 412 you check for none in if case.